### PR TITLE
Refactor common field absorber implementation

### PR DESCRIPTION
**Pull request by sbastrakov**
Move more responsibility into `fields::absorber::Absorber`.
Now more things are handled run-time.
This is to prepare the following transition to run-time absorber setting and unification of PML and non-PML field solvers.
User interface is unchanged yet.

### CODE REVIEW

**Comment by sbastrakov** [comment]
CI is failing with `E: The repository 'https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  Release' does not have a Release file.` for a few configurations. Hope it's a temporary glitch

**Review by PrometheusPi — APPROVED** [review]

**Comment by PrometheusPi** [comment]
@sbastrakov Have you run-time tested your code?

**Comment by sbastrakov** [comment]
@PrometheusPi i did.

**Comment by PrometheusPi** [comment]
@sbastrakov perfect :+1:

**Comment by PrometheusPi** [comment]
@ComputationalRadiationPhysics/picongpu-maintainers I will merge this pull request tomorrow if you do not oppose or need more time.

**Pull request merged**
